edit_course/operations/configure.py

```diff
@@ -497,22 +498,20 @@ def get_target_category(category, course=None, course_url=None):
 
     else:
 
-        course_slug_begin = course_url.find('/', course_url.find('://') + 1) + 1
+        parsed_url = urlparse(course_url)
+        service_hostname = urlparse(settings.BASE_URL).hostname
 
-        if settings.BASE_URL:
-            # Can correctly identify slug-beginning with BASE_URLs
-            # even if they end in '/'
-            course_slug_begin = course_url.find('/', len(settings.BASE_URL) - 1) + 1
+        if parsed_url.hostname != service_hostname:
+            return None, _("Course URL '{}' doesn't match service's hostname '{}'".format(course_url, service_hostname))
```

Comment:
You can't do formatting before translation is resolved the format *needs* to be `_("string").format(foo, bar)`. Furthermore, that would resolve the lazy string at this point, but we wan't to delay it until rendering. Thus, you really need to do `format_lazy(_("string"), foo, bar)`.  Sadly, format_lazy is not in a good place under lib. It's under `authorization/permissions.py` and `lib/remote_page.py`, but should be merged to single place e.g. `lib/translation.py`.